contracts/MockRule.sol

```diff
@@ -65,4 +65,49 @@ contract MockRule {
         value = _value;
         receivedPayableAmount = msg.value;
     }
+
+    /**
+     * @notice Mocks CoGateway redeem function.
+     *
+     * @param _amount Redeem amount that will be transferred form redeemer
+     *                account.
+     * @param _beneficiary The address in the origin chain where the value
+     *                     tok ens will be released.
+     * @param _facilitator Facilitator address.
+     * @param _gasPrice Gas price that redeemer is ready to pay to get the
+     *                  redemption process done.
+     * @param _gasLimit Gas limit that redeemer is ready to pay
+     * @param _nonce Nonce of the redeemer address.
+     * @param _hashLock Hash Lock provided by the facilitator.
+     *
+     * @return messageHash_ which is unique for each request.
+     */
+    function redeem(
+        uint256 _amount,
+        address _beneficiary,
+        address _facilitator,
+        uint256 _gasPrice,
+        uint256 _gasLimit,
+        uint256 _nonce,
+        bytes32 _hashLock
+    )
+        public
+        payable
+        returns (bytes32)
+    {
+        require(msg.value != uint256(0), "msg.value is 0.");
+
+        receivedPayableAmount = msg.value;
+        return keccak256(
```

Comment:
I don't think this mock redeem needs to hash anything. Further, if you stick with using `.call` to call CoGateway.redeem, I don't know that this function needs to return anything (or to have a declared return argument, for that matter), since it just needs to be a successful call, function signatures do not depend on return arguments, the return argument from the CoGateway.redeem does not appear to factor into TokenHolder.redeem or the tests.

Reply:
I agree with your point. Initially I implemented that way only.
However that generated ugly looking compilation warnings because the function arguments were not being used. This small trick solved warning issue 😉